python/semantic_kernel/skill_definition/sk_function_decorator.py

```diff
@@ -1,17 +1,23 @@
 # Copyright (c) Microsoft. All rights reserved.
 
 
-def sk_function(description: str):
+def sk_function(description: str, name: str = None, input_description: str = None, input_default_value: str = None):
     """
     Decorator for SK functions.
 
     Args:
         description -- The description of the function
+        name -- The name of the function
+        input_description -- The description of the input
+        input_default_value -- The default value of the input
     """
 
     def decorator(func):
         func.__sk_function__ = True
         func.__sk_function_description__ = description
+        func.__sk_function_name__ = name if name else func.__name__
```

Comment:
This is nice! I like that idea (just default to name of method). I don't know off the top of my head, but I think Python identifiers are all valid SK function names? (Might be worth doing a quick check of the regex we have and the Python grammar)

Reply:
btw, if you want to push this further, we could use a function comment for the function description, instead of the decorator

Reply:
![image](https://user-images.githubusercontent.com/371009/227345410-c9b3b7b0-b594-4ed2-9687-f3a0775fc25d.png)

the very first version of SK in python pulled that out, reducing the duplication and just asking a to document a function the usual way